src/com/oltpbenchmark/api/BenchmarkModule.java

```diff
@@ -318,7 +320,8 @@ public final void runScript(String script) {
                         throw new RuntimeException(msg, ex);
                     } finally {
                         for (LoaderThread t : loaderThreads) {
-                            t.getConnection().close();
+                        	t.getConnection().commit();
```

Comment:
Please guard `t.getConnection().commit();` with `if (!t.getConnection().getAutoCommit())`, i.e.:

```
if (!t.getConnection().getAutoCommit()) {
    t.getConnection().commit();
}
```

You may want to cache the connection.